lib/SILOptimizer/Utils/InstOptUtils.cpp

```diff
@@ -1800,15 +1800,22 @@ void swift::salvageDebugInfo(SILInstruction *I) {
     return;
 
   if (auto *SI = dyn_cast<StoreInst>(I)) {
-    if (SILValue DestVal = SI->getDest())
-      // TODO: Generalize this into "get the attached debug info
-      // on `DestVal`".
+    if (SILValue DestVal = SI->getDest()) {
       if (auto *ASI = dyn_cast_or_null<AllocStackInst>(
               DestVal.getDefiningInstruction())) {
-        if (auto VarInfo = ASI->getVarInfo())
+        if (auto VarInfo = ASI->getVarInfo()) {
+          SILLocation Loc = SI->getLoc();
+          bool IsValidLoc =
+            (Loc.isFilenameAndLocation() ?
+             Loc.getFilenameAndLocation()->line != 0 :
+             Loc.getSourceLoc().isValid());
```

Comment:
@adrian-prantl We're having the following:

```
SourceLoc SILLocation::getSourceLoc() const {
  if (isSILFile())
    return storage.sourceLoc;

  // Don't crash if the location is a FilenameAndLocation.
  // TODO: this is a workaround until rdar://problem/25225083 is implemented.
  if (getStorageKind() == FilenameAndLocationKind)
    return SourceLoc();

  return getSourceLoc(getPrimaryASTNode());
}
```

So in case of `isFilenameAndLocation() == true` we'd always end with empty source loc. I'm not 100% sure this is a real case here though :)